packages/web-components/src/components/ic-dialog/ic-dialog.tsx

```diff
@@ -117,46 +117,17 @@ export class Dialog {
   @Watch("open")
   watchOpenHandler(): void {
     if (this.open) {
-      this.dialogRendered = true;
-
-      if (this.disableHeightConstraint) {
-        this.dialogEl.show();
-      } else {
-        this.dialogEl.showModal();
-      }
-
-      setTimeout(() => {
-        this.fadeIn = true;
-
-        /**
-         * This is required to set scroll back to top if:
-         * - dialog content goes below the fold
-         * - is closed using cancel or confirm and reopened.
-         *
-         * Without this, the scroll bar will start from the dialog's last scroll-x coordinate.
-         */
-        if (this.disableHeightConstraint && this.backdropEl.scrollTop !== 0) {
-          this.backdropEl.scrollTop = 0;
-        }
-      }, 10);
-
-      setTimeout(() => {
-        this.setInitialFocus();
-        checkResizeObserver(this.runResizeObserver);
-      }, 75);
-
-      setTimeout(() => {
-        this.getFocusedElementIndex();
-        this.icDialogOpened.emit();
-      }, 80);
+      this.dialogOpened();
     } else {
       this.fadeIn = false;
       if (this.resizeObserver !== null) {
         this.resizeObserver.disconnect();
       }
       setTimeout(() => {
         this.dialogRendered = false;
-        this.dialogEl.close();
+        if (this.dialogEl?.close) {
```

Comment:
I do not think this if statement is required as `dialogEl` is always going to be a `<dialog>` element and the close method comes standard with it. 
ref: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog

Reply:
Agreed - this was left over from trying to getting this to work using a different lifecycle method. I will revert this change.